public/components/model_version/toggle_deploy_button.tsx

```diff
@@ -0,0 +1,278 @@
+                        <EuiButton
+                          color="danger"
+                          onClick={() => {
+                            const overlayRef = overlays?.openModal(
+                              <EuiModal onClose={() => overlayRef.close()}>
+                                <EuiModalHeader>
+                                  <EuiModalHeaderTitle>
+                                    <h2>
+                                      <EuiLink href={modelVersionUrl}>
+                                        {modelName} version {modelVersion}
+                                      </EuiLink>{' '}
+                                      deployment failed
+                                    </h2>
+                                  </EuiModalHeaderTitle>
+                                </EuiModalHeader>
+                                <EuiModalBody>
+                                  <EuiText size="s">Error message:</EuiText>
+                                  <EuiSpacer size="m" />
+                                  <EuiCodeBlock isCopyable={true}>
+                                    {res.error ?? 'Unknown error'}
+                                  </EuiCodeBlock>
+                                </EuiModalBody>
+                                <EuiModalFooter>
+                                  <EuiButtonEmpty onClick={() => overlayRef.close()}>
+                                    Close
+                                  </EuiButtonEmpty>
+                                </EuiModalFooter>
+                              </EuiModal>
+                            );
+                          }}
+                        >
+                          See full error
+                        </EuiButton>
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  </>
+                ),
+              },
+              { toastLifeTimeMs: 60000 }
+            );
+          }
+        },
+      });
+  };
+
+  const onUndeploy = async (id: string) => {
+    try {
+      setLoading(true);
+      await APIProvider.getAPI('model').unload(id);
+      notifications?.toasts.addSuccess({
+        title: mountReactNode(
+          <EuiText>
+            <EuiLink href={modelVersionUrl}>
+              {modelName} version {modelVersion}
+            </EuiLink>{' '}
+            has been undeployed
+          </EuiText>
+        ),
+      });
+      onComplete(modelVersionId);
+    } catch (e) {
+      onError(modelVersionId);
+      notifications?.toasts.addDanger(
+        {
+          title: mountReactNode(
+            <EuiText>
+              <EuiLink href={modelVersionUrl}>
+                {modelName} version {modelVersion}
+              </EuiLink>{' '}
+              undeployment failed
+            </EuiText>
+          ),
+        },
+        { toastLifeTimeMs: 60000 }
+      );
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const deployModal = isDeployModalVisible && (
+    <EuiConfirmModal
```

Comment:
Could we reuse `ModelVersionDeploymentConfirmModal` here?

Reply:
Created a hook `useDeployment` to abstract the deploy/undeploy logic